### PR TITLE
[SERVER-1666] 4.0 vm

### PR DESCRIPTION
**Pull request by lloydChris**
# Description
no KOTS anymore.  Add helm instructions for setting up VM service

# Reasons
A link to a GitHub and/or JIRA issue (if applicable).
Otherwise, a brief sentence about why you made these changes.

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: https://circleci.com/docs/style/.

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

File: jekyll/_cci2/server/Operator/vm-service.adoc [hunk 1]
```diff
@@ -35,39 +37,40 @@ There is also an option to change the port used for VM service. The default is `
...
+      subnets:
+      - <subnet-id>
+      securityGroupId: <security-group-id>
+
+      # Authenticate with IAM access keys
+      accessKey: <access-key>
+      secretKey: <secret-key>
+      # or IRSA (IAM roles for service accounts)
+      irsaRole: <role-arn>
+
```
Comment (ryanwohara) [inline jekyll/_cci2/server/Operator/vm-service.adoc:71]: I think we should include windowsAMI, linuxAMI, and tags here, as well.

File: jekyll/_cci2/server/Operator/vm-service.adoc [hunk 2]
```diff
@@ -93,41 +96,38 @@ The default AMIs for server v4.x are based on Ubuntu 20.04.
...
+      - circleci-vm
+      - <your-network>
+      zone: <zone>
+      network: <network>
+      subnetwork: <subnetwork>
+
+      service_account: <service-account-json>
+      # OR
+      workloadIdentity: ""  # Leave blank if using JSON keys of service account else service account email address
+
```
Comment (ryanwohara) [inline jekyll/_cci2/server/Operator/vm-service.adoc:128]: Can we also include windowsImage and linuxImage here?

File: jekyll/_cci2/server/Operator/vm-service.adoc [hunk 3]
```diff
@@ -93,41 +96,38 @@ The default AMIs for server v4.x are based on Ubuntu 20.04.
 ----
 
 === Google Cloud Platform
-You need the following fields to configure your VM service to work with Google Cloud Platform (GCP).
+You will need to add a section to your values.yaml file to configure your VM Service to work with AWS EC2.
```
Comment (atulsingh0) [inline jekyll/_cci2/server/Operator/vm-service.adoc:99]:
```suggestion
You will need to add a section to your values.yaml file to configure your VM Service to work with Google Cloud Platform (GCP). 
```

File: jekyll/_cci2/server/Operator/vm-service.adoc (hunk 1 above)
Comment (lloydChris) [reply jekyll/_cci2/server/Operator/vm-service.adoc:71]: My thought was to keep these instructions limited to a very "standard" example.  The values.yaml file itself can be the source of more advanced configuration options.

**Review by ryanwohara — APPROVED** [review]

**Review by atulsingh0 — APPROVED** [review]

File: jekyll/_cci2/server/Operator/vm-service.adoc [hunk 4]
```diff
@@ -11,13 +11,19 @@ version:
...
+CAUTION: We recommend that you leave these options at their defaults until you have successfully configured and verified the core and build services of your server installation. Steps to set up VM service are provided in the installation guide for  https://circleci.com/docs/2.0/server-3-install-build-services/#eks[AWS] and https://circleci.com/docs/2.0/server-3-install-build-services/#gke[GCP].
+
+WARNING: If https://circleci.com/docs/2.0/docker-layer-caching/[Docker Layer Caching (DLC)] is used, VM Service instances need to be spun up on demand. For this to happen, **either** ensure any preallocated instances are in use, **or** set both remote Docker and `machine` preallocated instance fields to `0`.
+
+CAUTION: When using preallocated instances be aware that a cron job is scheduled to cycle through these instances once per day to ensure they do not end up in an unworkable state.
 
 == VM service settings
 You need to provide the hostname/IP for your VM service load balancer:
```
Comment (rosieyohannan) [inline jekyll/_cci2/server/Operator/vm-service.adoc:29]: @lloydChris could you check this section? Does running this command give you the info needed to full in these parts of `values.yaml`:

```
region: <region>
      # Subnets must be in the same availability zone
      subnets:
      - <subnet-id>
      securityGroupId: <security-group-id>
```

Comment (lloydChris) [reply jekyll/_cci2/server/Operator/vm-service.adoc]: Good catch.  I removed the section, it is no longer nessisary

**Pull request merged**